src/c++/perf_analyzer/genai-perf/genai_perf/graphs/box_plot.py

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python3
+# Copyright 2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+
+import copy
+
+import pandas as pd
+import plotly.express as px
+from genai_perf.graphs.base_plot import BasePlot
+from genai_perf.llm_metrics import Statistics
+
+
+class BoxPlot(BasePlot):
+    """
+    Generate a box plot in jpeg and html format.
+    """
+
+    def __init__(self, stats: Statistics) -> None:
+        super().__init__(stats)
+
+    def create_box_plot(
```

Comment:
Since all derived plotting classes share this "create plot" behavior, I think we can come up with a common abstract base behavior
```python
class BasePlot:
    def create(self, x_key, y_key, title, filename_root, ...):
        raise NotImplementError
```
and each plot classes can override this method. For example,
```python
class BoxPlot(BasePlot):
    def create(...):
        # create box plot
```
This way we have a consistent way of creating a plot instead of each plot class having their own create method (which could be confusing and prone to mistakes).

Reply:
How about this api? 
```python
def create(
        self,
        x_key: str,
        y_key: str,
        x_metric: str,
        y_metric: str,
        graph_title: str,
        x_label: str,
        y_label: str,
        filename_root: str,
    ):
```
I think this covers all the plots and all the parameters seems generic enough to apply for all plots.

Reply:
The issue is this api does not allow for preprocessing the data which was necessary for the scatter plots.
I think, ultimately, you are correct. I just didnt know how to mesh an api like the one above with the idea of preprocessing data.

The best I could think of, was to pass in a dataframe along with the rest of the api. Then I can preprocess in the plot manager and build the dataframe beforehand. The plot only needs to know how to index the dataframe to get the data to plot

Does this seem more reasonable?

Reply:
I don't necessarily think building dataframe needs to happen outside of the plot (since it seems like it's pretty simple one liner) but all the other things like preprocessing should.